crates/hyperqueue/src/dashboard/ui/screens/worker/worker_info_table.rs

```diff
@@ -19,20 +18,11 @@ struct WorkerInfoDataRow {
 }
 
 impl WorkerInfoTable {
-    pub fn update(&mut self, data: &DashboardData, worker_id: Option<WorkerId>) {
-        self.table.set_items(vec![]);
-        if let Some(worker_id) = worker_id {
-            if data
-                .query_connected_worker_ids(SystemTime::now())
-                .any(|id| id == worker_id)
-            {
-                let info = data.query_worker_info_for(&worker_id);
-                if let Some(info) = info {
-                    let rows = create_rows(info);
-                    self.table.set_items(rows);
-                }
-            }
-        }
+    pub fn update(&mut self, data: &DashboardData, worker_id: WorkerId) {
```

Comment:
What about passing the `WorkerConfiguration` to this function directly? Then we could avoid the `unwrap` and handle it in the worker info screen.